sgxs-tools/src/sgx_detect/main.rs

```diff
@@ -248,6 +265,52 @@ impl SgxSupport {
             }
             Ok(Rc::new(RefCell::new(lib.build())))
         })();
+        let gsgxdev= match File::open("/dev/gsgx") {
+            Ok(_) => Ok(()),
+            Err(e) => Err(e.into()),
+        };
+        let nodeagent_status = (|| {
+            let request_url = format!("http://localhost:9092/v1/sys/version");
+            let mut response = reqwest::get(&request_url)?;
+            let ver :Version = response.json().unwrap();
+            Ok(NodeAgentStatus{current_ver: ver.version.clone()})
+        })();
+        let permdaemon_status = (|| -> Result<(), Error> {
+            let mut isgx = false;
+            let mut gsgx = false;
+            let mut installed = false;
+            let mut system = sysinfo::System::new();
+            system.refresh_all();
+            for proc in system.get_process_by_name("bash") {
+                if (proc.cmd().len() >= 2) && (installed == false) {
+                    installed = proc.cmd()[1].contains("sgx_perm_daemon");
```

Comment:
Is this really the best way to check this? How about `systemctl status perm_daemon` or so?

Reply:
will use `systemctl is-active sgx_perm_daemon`